syntax = "proto3";

package post;

message CreatePostRequest {
    string title = 1;
    string description = 2;
    string creator_id = 3;
    bool is_private = 4;
    repeated string tags = 5;
}

message UpdatePostRequest {
    string id = 1;
    string title = 2;
    string description = 3;
    string creator_id = 4;
    bool is_private = 5;
    repeated string tags = 6;
}

message Post {
  string id = 1;
  string title = 2;
  string description = 3;
  string creator_id = 4;
  string created_at = 5;
  string updated_at = 6;
  bool is_private = 7;
  repeated string tags = 8;
}

message PostRequest {
  string id = 1;
}

message PostList {
  repeated Post posts = 1;
}

message Empty {}


service PostService {
  rpc CreatePost(CreatePostRequest) returns (Post);
  rpc UpdatePost(UpdatePostRequest) returns (Post);
  rpc DeletePost(PostRequest) returns (Empty);
  rpc GetPost(PostRequest) returns (Post);
  rpc ListPosts(Empty) returns (PostList);
}